src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,138 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.CompletedCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.PendingCommand;
+import duke.command.ShowCommand;
+import duke.exception.DukeException;
+import duke.exception.InvalidArgumentException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidTaskTypeException;
+import duke.task.CommandType;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskType;
+import duke.task.ToDo;
+
+/**
+ * The Parser class parses and processes user input.
+ */
+public class Parser {
+    /**
+     * Parses the given command based on different keywords.
+     * Returns the appropriate {@link Command} to execute next.
+     * @param fullCommand A String read from user input.
+     * @return The corresponding {@link Command} for the input.
+     * @throws DukeException Exception when parsing the input.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```

Comment:
should this method be split into more methods? might be harder to debug in future :(